src/security/Handshake.cc

```diff
@@ -244,7 +244,11 @@ Security::HandshakeParser::parseModernRecord()
     Must(record.fragment.length() || record.type == ContentType::ctApplicationData);
 
     if (currentContentType != record.type) {
+        tkMessages.reinput(fragments, false);
+        parseMessages();
+
         Must(tkMessages.atEnd()); // no currentContentType leftovers
+
         fragments = record.fragment;
         tkMessages.reset(fragments, true); // true because more fragments may come
```

Comment:
This code used to be more-or-less logical -- we were preparing tkMessages for the unconditional parseMessages() call below. Now, it looks strange that we keep resetting tkMessages without actually using it. Same concern for the `else` part of this `if` statement.

Can you refactor to avoid pointless tkMessages resetting, perhaps by moving `tkMessages.reinput(fragments, false)` into parseMessages()? I have a feeling it would make the code simpler than it was before this PR.

Reply:
True.

commit 8c781bdfd